docs: third-party state management and migration guide (#19798)
nuxt/nuxt: docs/1.getting-started/7.state-management.md

---
navigation.icon: uil:database

> description: Nuxt powerful state management libraries and useState composable to create a reactive and SSR-friendly shared state.

Reviewer: The sentence now lacks a verb. Probably `provides` should stay insteand of being replaced? Or did you mean something else?

Author: you are right, I should have re-read my self

> description: Nuxt provides powerful state management libraries and useState composable to create a reactive and SSR-friendly shared state.

Reviewer 2: ```suggestion
description: Nuxt provides powerful state management libraries and the useState composable to create a reactive and SSR-friendly shared state.
```

---

# State Management

Nuxt **used to rely** on the Vuex library to provide global state management, if you are migrating from nuxt 2, please head to [migration guide](/docs/7.migration/2.configuration.md#vuex).

> Nuxt is now unopiniated about this topic, choose the right solution for your needs is up to you.

Reviewer 2: ```suggestion
Nuxt is not opiniated about state management, so feel free to choose the right solution for your needs.
```

> However here is some keys to help you make an informed decision.

Reviewer 2: ```suggestion

However here are some pointers to help you make an informed decision.
```

## Using a third-party library

> One of the features Nuxt can give you is a strong modules ecosystem. You will find multiple integrations with the most popular state management library such as:

Reviewer 2: ```suggestion
One of the strengths of Nuxt is its strong modules ecosystem. There are multiple integrations with the most popular state management libraries, including:

```

> - [Pinia](https://nuxt.com/modules/pinia) the official Vue recommendation.

Reviewer 2: ```suggestion
- [Pinia](https://nuxt.com/modules/pinia) - the official Vue recommendation
```

> - [Harlem](https://nuxt.com/modules/harlem) immutable global state management.
> - [XState](https://nuxt.com/modules/xstate) approches state as machine and provides visual tools.

Reviewer 2: ```suggestion
- [Harlem](https://nuxt.com/modules/harlem) - immutable global state management
- [XState](https://nuxt.com/modules/xstate) - state machine approach with tools for visualising and testing your state logic
```

> ## Using useState

Reviewer 2: ```suggestion
## Using `useState`
```

> Nuxt provides useState composable to create a reactive and SSR-friendly shared state across components.

Reviewer 2: ```suggestion
Nuxt provides the `useState` composable to create a reactive and SSR-friendly shared state across components.
```
